core/service/src/engine/context.rs

```diff
@@ -81,11 +113,13 @@ pub enum NodeInfoVersioned {
 #[derive(Clone, Debug, Versionize, Serialize, Deserialize)]
 #[versionize(NodeInfoVersioned)]
 pub struct NodeInfo {
-    pub name: String,
+    pub mpc_identity: String,
     pub party_id: u32,
-    pub verification_key: PublicSigKey,
-    pub backup_encryption_public_key: UnifiedPublicEncKey,
+    /// This is optional for legacy reasons because typically MPC parties
+    /// do not know the public verification keys of other parties when it first starts.
+    pub verification_key: Option<PublicSigKey>,
```

Comment:
I don't think `NodeInfo` has been used before so I don't think we need to do anything for legacy reasons here. Or maybe I am missing something?

Reply:
legacy as in we still need to support starting MPC parties from a config file. but in the config file we do not specify the public signing key of the other parties.

Reply:
So am I understanding correctly that once we are done implementing context then it will no longer be needed? I don't think we initially need to load form file. At least as I see it once we setup everything the servers will boot without a context and will need an initial context before they can be used for anything